server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -249,8 +311,39 @@ SearchBackpressureSettings getSettings() {
         return settings;
     }
 
-    SearchBackpressureState getState() {
-        return state;
+    SearchBackpressureState getSearchBackpressureStats(Class<? extends SearchBackpressureTask> taskType) {
+        return searchBackpressureStates.get(taskType);
+    }
+
+    /**
+     * Given the threshold suppliers, returns the list of applicable trackers
+     */
+    public static List<TaskResourceUsageTracker> getTrackers(
+        LongSupplier cpuThresholdSupplier,
+        DoubleSupplier heapVarianceSupplier,
+        LongSupplier heapBytesThresholdSupplier,
+        int heapMovingAverageWindowSize,
+        LongSupplier ElapsedTimeNanosSupplier,
+        ClusterSettings clusterSettings,
+        Setting<Integer> windowSizeSetting
+    ) {
+        List<TaskResourceUsageTracker> trackers = new ArrayList<>();
+        trackers.add(new CpuUsageTracker(cpuThresholdSupplier));
+        if (HEAP_SIZE_BYTES > 0) {
```

Comment:
You could have this check on `HeapUsageTracker` as a static method, fe `isHeapTrackingSupported`. that would clearly communicate what is the purpose:
```
if (HeapUsageTracker.isHeapTrackingSupported()) {
 ...
}
```

Reply:
Makes sense. Updated in the next commit.